### PR TITLE
Us101463/refactor module link

### PR DESCRIPTION
**Pull request by tinglisD2L**

### CODE REVIEW

**Comment by tinglisD2L** [comment]
@jwong-d2l Can't figure out how to remove :hover styling from outer-module and inner-module entirely, but at least the styling looks visually correct now and things are refactored.

File: components/d2l-inner-module.html [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+				background-color: var(--d2l-color-celestine-plus-2);
+			}
+
+		</style>
+
+		<div
+			class$="[[_getIsSelected(selectedModule)]]"
+			on-click="toggleSelected"
+		>
+			<div class="header-container">
```
Comment (jwong-d2l) [inline components/d2l-inner-module.html]: are both divs necessary here?

Comment (jwong-d2l) [reply components/d2l-inner-module.html]: or any divs?  Can we remove both and target the span?

File: demo/index.html [hunk 2]
```diff
@@ -38,11 +38,11 @@ <h4>Sequence navigator in a sidebar</h4>
 			<demo-snippet>
 				<template>
 					<h4>Module</h4>
-					<d2l-module-link
+					<d2l-outer-module
```
Comment (jwong-d2l) [inline demo/index.html]: did you want to make an inner module one here too?

File: test/d2l-inner-module.html [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+				describe('for module with all optional activities unread', () => {
+
+					let element;
+
+					beforeEach(async() => {
+						element = await SirenFixture.load('data/l4-optional-completion-module.json', fixture('Empty'));
+					});
+
+					it('should show only the title in the header, when the accordion is open', () => {
```
Comment (jwong-d2l) [inline test/d2l-inner-module.html]: Are there accordions in the inner module?

File: components/d2l-inner-module.html (hunk 1 above)
Comment (tinglisD2L) [reply components/d2l-inner-module.html]: hmm it's not easily removable... currently the inner span is used to do the multiline truncation, the middle div is used for the table layout, and the outer div is needed for the hover effect. I tried removing some of them and couldn't figure out to restyle it to look the same

File: test/d2l-inner-module.html (hunk 3 above)
Comment (tinglisD2L) [reply test/d2l-inner-module.html]: oops!

File: demo/index.html (hunk 2 above)
Comment (tinglisD2L) [reply demo/index.html]: sure

File: components/d2l-inner-module.html (hunk 1 above)
Comment (jwong-d2l) [reply components/d2l-inner-module.html]: Ok maybe we can try again later

**Pull request merged**